File: Leetcode/Medium/49. Group Anagrams.cpp
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        mapping = {}
        for word in strs:
            key = ''.join(sorted(word))
            if key in mapping:
                mapping[key].append(word)
            else: mapping[key] = [word]
        return list(mapping.values())

File: Leetcode/Medium/49. Group Anagrams/49. Group Anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagrams = {}

        for string in strs:
            freqs = [0] * 26
            
            for c in string:
                freqs[ord(c) - ord('a')] += 1
            
            freqs = tuple(freqs)
            if freqs in anagrams:
                anagrams[freqs].append(string)
            else:
                anagrams[freqs] = [string]
        res = []
        for anagram in anagrams.values():
            res.append(anagram)

        return res
                
